From 0d0876baca8e87305b25fb8ee640f683d9fc410a Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai@quantatw.com>
Date: Wed, 25 Mar 2026 20:05:24 +0800
Subject: [PATCH 07/10] configurations: nvidia: clover_cx9: add NIC sensor
 configs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add NIC sensor configuration to enable sensor monitoring.

[1] Gerrit: https://gerrit.openbmc.org/c/openbmc/entity-manager/+/88679

Tested:
```
root@bmc:~# busctl tree xyz.openbmc_project.EntityManager | grep -e "NS_NIC"
          │ ├─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_Clover/NS_NIC_OSFP_TEMP_C
          │ ├─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_Clover/NS_NIC_PWR_W
          │ ├─ /xyz/openbmc_project/inventory/system/board/NVIDIA_VR_NVL72_Clover/NS_NIC_TEMP_C
```

Signed-off-by: Potin Lai <potin.lai@quantatw.com>
Change-Id: I3fc20fb3a25483d6aaaa72a4da41d345d15bd0c4
---
 configurations/nvidia/clover_cx9.json | 43 +++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/configurations/nvidia/clover_cx9.json b/configurations/nvidia/clover_cx9.json
index 1129750..4deaec3 100644
--- a/configurations/nvidia/clover_cx9.json
+++ b/configurations/nvidia/clover_cx9.json
@@ -5,6 +5,49 @@
             "Bus": "$bus",
             "Name": "NVIDIA VR NVL72 Clover FRU",
             "Type": "EEPROM"
+        },
+        {
+            "MaxValue": 255,
+            "MinValue": -255.0,
+            "Name": "NS_NIC_TEMP_C",
+            "PowerState": "On",
+            "Timeout": 30.0,
+            "Type": "ExternalSensor",
+            "Units": "DegreesC"
+        },
+        {
+            "MaxValue": 255,
+            "MinValue": -255.0,
+            "Name": "NS_NIC_OSFP_TEMP_C",
+            "PowerState": "On",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.36,
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 68
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.1,
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 5
+                }
+            ],
+            "Timeout": 30.0,
+            "Type": "ExternalSensor",
+            "Units": "DegreesC"
+        },
+        {
+            "MaxValue": 65535,
+            "MinValue": 0,
+            "Name": "NS_NIC_PWR_W",
+            "PowerState": "On",
+            "Timeout": 30.0,
+            "Type": "ExternalSensor",
+            "Units": "Watts"
         }
     ],
     "Name": "NVIDIA VR NVL72 Clover",
-- 
2.52.0

